From 38de4bccdb8a861ffdd447f12fdab19d6d852c02 Mon Sep 17 00:00:00 2001
From: Chong Lu <Chong.Lu@windriver.com>
Date: Tue, 26 Jun 2018 17:34:15 +0800
Subject: [PATCH] vim: add knob whether elf.h are checked

Previously, it still was checked when there was no elf library in sysroots directory.
Add knob to decide whether elf.h are checked or not.

Upstream-Status: Pending

Signed-off-by: Chong Lu <Chong.Lu@windriver.com>
Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 src/configure.ac | 7 +++++++
 1 file changed, 7 insertions(+)

Index: git/src/configure.ac
===================================================================
--- git.orig/src/configure.ac
+++ git/src/configure.ac
@@ -3264,11 +3264,18 @@ AC_TRY_COMPILE([#include <stdio.h>], [in
 	AC_MSG_RESULT(no))
 
 dnl Checks for header files.
+AC_MSG_CHECKING(whether or not to look for elf.h)
+AC_ARG_ENABLE(elf-check,
+        [  --enable-elf-check      If elfutils, check for elf.h [default=no]],
+        , enable_elf_check="no")
+AC_MSG_RESULT($enable_elf_check)
+if test "x$enable_elf_check" != "xno"; then
 AC_CHECK_HEADER(elf.h, HAS_ELF=1)
 dnl AC_CHECK_HEADER(dwarf.h, SVR4=1)
 if test "$HAS_ELF" = 1; then
   AC_CHECK_LIB(elf, main)
 fi
+fi
 
 AC_HEADER_DIRENT
 
